### PR TITLE
Clean up uart0 devicetree, don't enable by default

### PR DESCRIPTION
**Pull request by petejohanson**
* Set pro-micro TX/RX pins, but don't enable the UART
  by default.

### CODE REVIEW

File: app/boards/arm/nrf52840_m2/nrf52840_m2.dts [hunk 1]
```diff
@@ -51,16 +51,6 @@
...
 
-&uart0 {
-	compatible = "nordic,nrf-uarte";
-	status = "okay";
-	current-speed = <115200>;
-	tx-pin = <19>;
-	rx-pin = <21>;
-	rts-pin = <23>;
-	cts-pin = <25>;
-};
```
Comment (mcrosson) [inline app/boards/arm/nrf52840_m2/nrf52840_m2.dts:62]: According to the makerdiary docs / stock dts file (apache 2.0 licensed) the nrf52840 has a uart and has the following definition per the rest of this PR.

```
&uart0 {
	compatible = "nordic,nrf-uart";
	current-speed = <115200>;
	tx-pin = <16>;
	rx-pin = <15>;
	//rts-pin = <14>;
	//cts-pin = <13>;
};
```

Comment (petejohanson) [reply app/boards/arm/nrf52840_m2/nrf52840_m2.dts:62]: Oh. I see that when plugged into https://wiki.makerdiary.com/nrf52840-m2-devkit/ dev kit, those are the pins they use. That seems to me to be something you'd add to the devkit definition, not to this. one.

Comment (mcrosson) [reply app/boards/arm/nrf52840_m2/nrf52840_m2.dts:62]: Agreed, closing review / comment.

File: app/boards/arm/nice_nano/nice_nano.dts [hunk 2]
```diff
@@ -65,12 +65,9 @@
 
 &uart0 {
 	compatible = "nordic,nrf-uarte";
-	status = "okay";
 	current-speed = <115200>;
```
Comment (Nicell) [inline app/boards/arm/nice_nano/nice_nano.dts:68]:
```suggestion
```

**Review by Nicell — CHANGES_REQUESTED** [review]

File: app/boards/arm/nice_nano/nice_nano.dts (hunk 2 above)
Comment (petejohanson) [reply app/boards/arm/nice_nano/nice_nano.dts]: Fixed. Good catch.

**Review by Nicell — APPROVED** [review]

**Pull request merged**